Library/Homebrew/unpack_strategy/p7zip.rb

```diff
@@ -16,7 +16,7 @@ def self.can_extract?(path)
       path.magic_number.match?(/\A7z\xBC\xAF\x27\x1C/n)
     end
 
-    sig { returns(T.nilable(T::Array[Formula])) }
+    sig { returns(T::Array[Formula]) }
     def dependencies
       @dependencies ||= T.let([Formula["p7zip"]], T.nilable(T::Array[Formula]))
```

Comment:
@samford I'm out of my depth a bit here, so if there's a more comprehensive solution, I'm all for you opening an alternative and can close this.

Reply:
I tried a couple of simple approaches earlier and didn't meet with any success (I've wrestled with `must be declared inside initialize...` errors in other areas before and it was like banging my head against a wall). I imagine someone more knowledgeable than me could probably figure something out.

I'm fine with this, if we also remove `T.nilable` from the `#dependencies` `sig` for the other strategies. Since the `UnpackStrategy#dependencies` signature doesn't allow for a nilable type (as requested), the other strategies' `#dependencies` methods will produce a runtime error unless they're updated like `P7Zip`.